### ISSUE TITLE
bogus or missing JSC_USED_GLOBAL_THIS warning with ADVANCED_OPTIMIZATIONS

### ISSUE THREAD

**Issue by blickly** [google/closure-compiler#279]
This issue was imported from Closure Compiler's previous home at http://closure-compiler.googlecode.com

The original discussion is archived at:
http://blickly.github.io/closure-compiler-issues/#909

**Comment by Dominator008** [comment]
Repro:
```js
// warning
document.body.onclick = function() {
    this.removeAttribute('style');
};

// no warning
document.body.onclick = function() {
    (function(a) {
        a.removeAttribute('style');
    })(this);
};
```

**Comment by ChadKillingsworth** [comment]
Honestly this specific case just means we need to improve the extern type information. The type of `this` should be inferred by both OTI and NTI.

https://github.com/google/closure-compiler/blob/master/externs/browser/w3c_dom1.js#L671-L707

**Comment by dimvar** [comment]
The global-this warning warns only on property accesses, that's why it doesn't warn for the second case. And for the first case, I think it warns because the assignment isn't a general method declaration (on a prototype property) but rather a one-off assignment to an object, and I'm guessing the pattern isn't recognized.
